cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/StatefulSetOperator.java

```diff
@@ -290,4 +291,31 @@ protected StatefulSetDiff revertStorageChanges(StatefulSet current, StatefulSet
         );
         return result;
     }
+
+    protected Future<String> getUid(String namespace, String podName) {
+        Future<String> result = Future.future();
+        vertx.createSharedWorkerExecutor("kubernetes-ops-tool").executeBlocking(
+            future -> {
+                String uid = getPodUid(podOperations.get(namespace, podName));
+                future.complete(uid);
+            }, true, result.completer()
+        );
+        return result;
+    }
+
+    private static String getPodUid(Pod resource) {
+        if (resource == null) {
+            return NO_UID;
+        }
+        return getUid(resource.getMetadata());
+    }
+
+    private static String getUid(ObjectMeta objectMeta) {
```

Comment:
I don't think there's much benefit in having both getPodUid() and getUid(). They could be merged into a single `getUid(HasMetadata)` method.